### ISSUE TITLE
Update the`mteb/stsbenchmark-sts` dataset

### ISSUE THREAD

**Issue by KennethEnevoldsen** [embeddings-benchmark/mteb#4798]
### Description of the feature

Originally posted [here](https://huggingface.co/datasets/mteb/stsbenchmark-sts/discussions/2) by [vlhandfo](https://huggingface.co/vlhandfo)

When working with this dataset, I found that there are duplicated instances in this dataset, including sentence pairs that appear in multiple partitions. There are also 11 instances where the sentence pairs are the same, but the scores are different. I have included my EDA code below:

``` {python}
from datasets import load_dataset
import pandas as pd

dataset = load_dataset("mteb/stsbenchmark-sts")

# Convert the dataset to a pandas DataFrame for easier manipulation
train_df = pd.DataFrame(dataset['train'])
dev_df = pd.DataFrame(dataset['validation'])
test_df = pd.DataFrame(dataset['test'])

# Combine all splits into a single DataFrame for duplication analysis
all_df = pd.concat([train_df, dev_df, test_df], ignore_index=True).reset_index(drop=True)

duplicated_sentence_pairs = all_df[all_df[['sentence1', 'sentence2']].duplicated(keep=False)]
# display(duplicated_sentence_pairs.sort_values(by=['sentence1', 'sentence2']))
num_duplicated_pairs = duplicated_sentence_pairs[['sentence1', 'sentence2']].drop_duplicates().shape[0]
print("Number of (sentence1, sentence2) pairs appearing in multiple times:", num_duplicated_pairs)
# >> 55

# keep duplicated sentence pairs that appear in more than one split
duplicated_cross_split = duplicated_sentence_pairs[
    duplicated_sentence_pairs.groupby(['sentence1', 'sentence2'])['split'].transform('nunique') > 1
].sort_values(by=['sentence1', 'sentence2', 'split'])
num_unique_pairs = duplicated_cross_split[['sentence1', 'sentence2']].drop_duplicates().shape[0]
print("Number of (sentence1, sentence2) pairs appearing in multiple splits:", num_unique_pairs)
# >> 15

contamination_counts = {}
for i, group in duplicated_cross_split.groupby(['sentence1', 'sentence2']):
    contamination_type = tuple(set(group["split"].tolist()))
    contamination_counts[contamination_type] = contamination_counts.get(contamination_type, 0) + 1

print("Number of different types of contamination:", contamination_counts)
# >> {('train', 'dev'): 6, ('test', 'train'): 7, ('test', 'dev'): 2}

same_score = duplicated_sentence_pairs[duplicated_sentence_pairs[['sentence1', 'sentence2', 'score']].duplicated(keep=False)].sort_values(by=['sentence1', 'sentence2'])

rows_not_in_same_score = duplicated_sentence_pairs.loc[duplicated_sentence_pairs.index.difference(same_score.index)].sort_values(by=['sentence1', 'sentence2'])
print("Instances of the same sentence pairs with different scores:", len(rows_not_in_same_score) // 2)
# >> 11
```

**Comment by KennethEnevoldsen** [comment]
Hi vlhandfo,

MTEB currently has tests to ensure that any newly added dataset contains no duplicates either across splits, within splits, or with different labels. However, for the existing dataset, it can be hard to change them, as it requires updating the benchmarks and thereby invalidating all previous evaluations, making scores incomparable.

The solution we typically use is to introduce a v2 ([example](https://embeddings-benchmark.github.io/mteb/overview/available_tasks/classification/#poemsentimentclassificationv2)), which we will then add to the benchmark upon its next update.

We do record [descriptive statistics](https://embeddings-benchmark.github.io/mteb/whats_new/?h=descr#descriptive-statistics) on the tasks, e.g. number of unique texts and number of samples from which you can derive duplicates.

For this dataset, you can also see this [here](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/descriptive_stats/STS/STSBenchmark.json), where we also see that for the test set, there is only 1 duplicate.

Though it is probably worth also removing the samples that appear in the training set.

**Comment by vlhandfo** [comment]
Hi,

Thanks for the information. My comment was mostly meant to let you all know what I had found, if you did not already know. 

I agree that it would be worth removing instances that appear in the training set from the validation sets. Something that I wanted to highlight, that I probably did not convey clearly enough in the first post, is that there are a few instances (11) of duplicate sentence pairs _that do not have the same score_. 7 of the duplicates appear in different partitions, but 4 instances occur in the train split. It seems that the sentence pairs were used in multiple years, but given different scores. It isn't a big issue, and the scores all fall 4.0 and up, but I thought I would mention it.

<img width="1011" height="231" alt="Image" src="https://github.com/user-attachments/assets/5384a5e7-6d9f-45e6-b36a-863c724d65b1" />

**Comment by KennethEnevoldsen** [comment]
Yes these should probably be removed. In general great to see people report issues 

I will mark this an issue as "good for newcomers". I don't think it takes priority, but it is a good starting point for people wanting to contribute.

**Comment by GioiaZheng** [comment]
This looks like a good `v2` case rather than something to mutate in-place on `STSBenchmark`.

The implementation boundary I would use:

- keep the current `STSBenchmark` task as-is for score comparability;
- create a cleaned HF dataset revision/path for the corrected split;
- add `STSBenchmark.v2` with `adapted_from=["STSBenchmark"]`;
- mark the original task with `superseded_by="STSBenchmark.v2"` once the v2 task exists;
- refresh descriptive stats so the before/after duplicate counts are visible.

For the cleaning rule, I would treat sentence-pair identity as a dataset-quality invariant, not just row identity:

- no `(sentence1, sentence2)` pair in validation/test should appear in train;
- no duplicate pair should remain inside an eval split;
- pairs with conflicting scores should be removed or resolved by an explicit documented policy;
- I would also check swapped order `(sentence2, sentence1)` because STS is semantically symmetric, even if the current report only counts same-order duplicates.

That keeps old leaderboard numbers comparable while giving new evaluations a clean target. If this split sounds right, I can take a pass at the MTEB-side PR once the cleaned dataset path/revision is available; otherwise I can start by producing the cleaning script + exact removal count table.

**Comment by GioiaZheng** [comment]
I ran a quick count against the current HF `train.jsonl.gz` / `validation.jsonl.gz` / `test.jsonl.gz` files to make the cleaning boundary more concrete.

Split sizes:

- train: 5,749
- validation: 1,500
- test: 1,379
- total: 8,628

Using exact `(sentence1, sentence2)` identity, I get the same shape as the original report:

- duplicate pair keys: 55
- duplicate rows involved: 116
- cross-split duplicate pair keys: 15
- eval/train overlap pair keys: 13
- conflicting-score pair keys: 11
- conflicting-score keys that cross splits: 7
- cross-split types: `test+train: 7`, `test+validation: 2`, `train+validation: 6`

If STS pair identity is treated as order-insensitive, which I think is worth checking because the task is semantically symmetric, the counts are higher:

- duplicate pair keys: 67
- duplicate rows involved: 150
- cross-split duplicate pair keys: 20
- eval/train overlap pair keys: 18
- conflicting-score pair keys: 25
- conflicting-score keys that cross splits: 12
- cross-split types: `test+train: 11`, `test+train+validation: 1`, `test+validation: 2`, `train+validation: 6`

So I think the v2 cleaning policy should explicitly decide whether swapped sentence order is considered the same pair. My instinct is yes for STS, but if the goal is the smallest possible compatibility-preserving fix, the exact-order rule reproduces the original issue report.
